chain/sync.go

```diff
@@ -610,6 +623,19 @@ func (s *Syncer) blockConnected(ctx context.Context, params json.RawMessage) err
 		return err
 	}
 
+	// Ensure the ancestor is in the main chain.  If it is not, this means
+	// we missed some blocks and should perform a new round of initial
+	// header sync.
+	if inMainChain, _, _ := s.wallet.BlockInMainChain(ctx, &header.PrevBlock); !inMainChain {
+		if !inMainChain {
+			log.Infof("Received header for block %s (height %d) when "+
+				"parent %s not in main chain. Re-requesting "+
+				"missing headers.", header.BlockHash(),
+				header.Height, header.PrevBlock)
+			return s.getHeaders(ctx)
```

Comment:
This is pretty much the only significant change on this PR.

Given we are moving the `notifyblocks` request **after** `getHeaders` returns during initial sync, there's a chance that we could miss blocks. So if the node sends a block for which we don't have its parent, we restart the `getHeaders` from scratch to fetch the missing blocks (including the one in this ntfn).